ShadowDOM/explainer.md

```diff
@@ -276,8 +274,160 @@ Earlier versions of this document used the `<script>` tag for declaring CSS Modu
+
+Another option is to instead use existing HTML concepts for applying stylesheets into shadow roots, such as the `<link>` tag, as demonstrated by the following example:
+
+```html
+<style type="module" specifier="foo">
+  #content {
+    color: red;
+  }
+</style>
+
+<my-element>
+  <template shadowrootmode="open">
+    <link rel="adoptedstylesheet" specifier="foo">
+  </template>
+</my-element>
+```
+
+While this approach doesn't map as closely to the existing `adoptedStyleSheets` API, it more closely follows existing HTML semantics. It also allows for a rich set of [features](https://html.spec.whatwg.org/#the-link-element) offered by the `<link>` element, such as media queries.
+
+The `adoptedstylesheets` attribute as specified accepts a list a stylesheets. Multiple stylesheets can be added to a shadow root's adoped stylesheet list with the `<link>` proposal by including multiple `<link>` tags.
+
+Looking forward, the `<link>` approach directly compatible with the proposed CSS `@sheet` feature, which allows a single CSS file to contain multiple stylesheets. This allows developers to specify a single named stylesheet that is applied from the CSS definition, rather than applying the global contents of the entire sheet.
+
+### Detailed Parsing Workflow
+
+In the following example:
+
+```html
+<style type="module" specifier="foo">
+  #content {
+    color: red;
+  }
+</style>
+<my-element>
+  <template shadowrootmode="open" adoptedstylesheets="foo">
+    ...
+  </template>
+</my-element>
+```
+
+Upon parsing the `<style>` tag above, an entry is added to the module map with the specifier of "foo" whose contents is a `CSSStyleSheet`, following the same steps as [create a CSS module script](https://html.spec.whatwg.org/#creating-a-css-module-script).
+
+As with existing `<style>` tags, if the CSS contains invalid syntax, error handling follows the rules specified in [error handling](https://www.w3.org/TR/css-syntax-3/#error-handling).
+
+When the `<template>` element is constructed, the `adoptedstylesheets` attribute is evaluated. Each space-separated identifier in the attribute is looked up in the module map. If an entry with that specifier
+exists in the module map, the associated `CSSStyleSheet` is added to the `adoptedStyleSheets` backing list associated with the `<template>` element's [shadow root](https://www.w3.org/TR/cssom-1/#dom-documentorshadowroot-adoptedstylesheets) in specified order, as defined in [CSS Style Sheet Collections](https://www.w3.org/TR/cssom-1/#css-style-sheet-collections).
+If an entry with that specifier does not exist in the module map, an empty `CSSStyleSheet` object is inserted into the module map with the specified `specifier`.
+
+This may also happen in reversed order, as in the following example:
+
+```html
+<my-element>
+  <template shadowrootmode="open" adoptedstylesheets="foo">
+    ...
+  </template>
+</my-element>
+<style type="module" specifier="foo">
```

Comment:
I think a framing of this example that will come up more commonly in practice is to have the `<style>` element as a child of the `<template>`.

Reply:
Good point. Added another example.